### PR TITLE
feat(app): implement send protocol to OT-3

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

adds a slideout that allows a user to send a protocol file to an OT-3, without also creating a run. refactors the choose robot slideout to share code. adds utilities to differentiate between intended robot for a protocol

closes RCORE-302

# Changelog

 - Implements send protocol to OT-3

# Review requests

smoke test both choose robot slideouts, send protocol and start run

# Risk assessment

low

### CODE REVIEW

File: shared-data/protocol/types/schemaV7/index.ts [hunk 1]
```diff
@@ -94,6 +98,7 @@ export interface ProtocolAnalysisOutput {
   modules: LoadedModule[]
   liquids: Liquid[]
   errors: AnalysisError[]
+  robotType?: RobotType
```
Comment (brenthagen) [inline shared-data/protocol/types/schemaV7/index.ts:101]: @shlokamin @smb2268 this was recently added to schemaV6, it should go here as well, right?

File: app/src/organisms/ChooseRobotToRunProtocolSlideout/AvailableRobotOption.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            height: 3.5625rem;
+          `}
+        />
+        <Flex
+          flexDirection={DIRECTION_COLUMN}
+          marginLeft={SPACING.spacing4}
+          marginTop={SPACING.spacing3}
+          marginBottom={SPACING.spacing4}
+        >
+          <StyledText as="h6">{robotModel}</StyledText>
```
Comment (koji) [inline app/src/organisms/ChooseRobotToRunProtocolSlideout/AvailableRobotOption.tsx]: probably need to add `fontWeight={TYPOGRAPHY.fontWeightSemiBold}`

File: app/src/organisms/ChooseRobotToRunProtocolSlideout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  showSlideout: boolean
+}
+
+export function ChooseRobotToRunProtocolSlideoutComponent(
+  props: ChooseRobotToRunProtocolSlideoutProps
+): JSX.Element | null {
+  const { t } = useTranslation(['protocol_details', 'shared', 'app_settings'])
+  const { storedProtocolData, showSlideout, onCloseClick } = props
+  const history = useHistory()
+  const [shouldApplyOffsets, setShouldApplyOffsets] = React.useState(true)
```
Comment (koji) [inline app/src/organisms/ChooseRobotToRunProtocolSlideout/index.tsx]:
```suggestion
  const [shouldApplyOffsets, setShouldApplyOffsets] = React.useState<boolean>(true)
```

File: app/src/organisms/ProtocolDetails/RobotConfigurationDetails.tsx [hunk 4]
```diff
@@ -109,8 +110,13 @@ export const RobotConfigurationDetails = (
     <Flex flexDirection={DIRECTION_COLUMN} paddingBottom={SPACING.spacing5}>
       <RobotConfigurationDetailsItem
         label={t('robot')}
-        // TODO(bh, 2022-10-18): read intended robot from protocol
-        item={isLoading ? loadingText : <StyledText as="p">OT-2</StyledText>}
+        item={
+          isLoading ? (
+            loadingText
+          ) : (
+            <StyledText as="p">{getRobotTypeDisplayName(robotType)}</StyledText>
```
Comment (koji) [inline app/src/organisms/ProtocolDetails/RobotConfigurationDetails.tsx:117]: From the design Mel put the version next to the `robotType`.
But not sure that is robot server version or something else and the design config itself should be for an OT-3 (there is a Gripper but robotType is OT-2...)

**Review by koji — APPROVED** [review]
I left a couple of comments but overall, the changes look good to me.

File: shared-data/protocol/types/schemaV7/index.ts (hunk 1 above)
Comment (smb2268) [reply shared-data/protocol/types/schemaV7/index.ts:101]: It looks like @SyntaxColoring added `robotType` to this analysis output in https://github.com/Opentrons/opentrons/pull/11610/files. I think it's correct to add it here but I don't know if it is an optional field on the server side - it looks required. 
Protocol analysis output is not strictly aligned with schema versioning so I don't know if it's technically correct to have this in schemaV6 types.. Any fields that are added on the python side will only be available in analysis if the robot server is updated and the file is reanalyzed after the update. We're hoping to handle this with the [Stale Data Recovery epic](https://opentrons.atlassian.net/browse/RAUT-279) down the line.

Comment (smb2268) [reply shared-data/protocol/types/schemaV7/index.ts:101]: This doesn't need to block merging, I'll make a note in the stale data epic and maybe we should think about moving this type out of the schema specific type folders.

File: app/src/organisms/ProtocolDetails/RobotConfigurationDetails.tsx (hunk 4 above)
Comment (koji) [reply app/src/organisms/ProtocolDetails/RobotConfigurationDetails.tsx:117]: @brenthagen 
Jesse, Emily, and Mel had a conversation on this and they decided to remove it.
So please ignore this

https://www.figma.com/file/0hYQ4lFJbAAI33jQMRt1Di/Opentrons-App-w-OT-3?node-id=1388%3A36044#329901251

**Pull request merged**